marcdown_project/marcdown/models/note.py

```diff
@@ -3,35 +3,65 @@
 import diff_match_patch as dmp_module
 import re
 
+from .tag import Tag
 
 class Note(models.Model):
     owner = models.ForeignKey(
-        'Profile', on_delete=models.CASCADE, related_name='ownNotes')
+        'Profile', on_delete=models.CASCADE, related_name='own_notes')
     public = models.BooleanField(default=True)
     sharers = models.ManyToManyField(
-        'Profile', blank=True, related_name='sharedNotes')
+        'Profile', blank=True, related_name='shared_notes')
 
     read_only = models.BooleanField(default=False)
 
     title = models.TextField()
-    tags = models.ManyToManyField('Tag', blank=True)
+    tags = models.ManyToManyField(Tag, blank=True)
     content = models.TextField()
 
     def __str__(self):
         return self.title
+    
+    def allow_reading_by_user(self, profile):
+        '''
+        Returns whether the given user has read permission on the note
+        '''
+        if profile == self.owner:
+            return True
+        else:
+            return self.public or profile in self.sharers.all()
 
-    def update(self, diff):
+    def allow_update_from_user(self, profile):
+        '''
+        Returns whether the given user has write permission on the note
         '''
-        Updates the content of the note using the given diff
+        if profile == self.owner:
```

Comment:
Readonly != "locked" right? We should probably create an issue for locking notes

Reply:
ReadOnly means only the owner has the right to edit, right ?

Reply:
Yeah, we'll add concept of locking a note later if we have time